test/FunctionsV2/StorageProviderSelectionTests.cs

```diff
@@ -4,77 +4,174 @@
+            DurableTaskOptions options = new DurableTaskOptions();
+            options.StorageProvider["type"] = storageProvider;
 
-        public static IEnumerable<object[]> ValidStorageProviderTypesData { get; }
+            using (ITestHost host = TestHelpers.GetJobHostWithMultipleDurabilityProviders(
+                options: options,
+                durabilityProviderFactories: durabilityProviderFactories))
+            {
+                azureStorageMock.Verify(a => a.GetDurabilityProvider(), Times.Never());
+                netheriteMock.Verify(n => n.GetDurabilityProvider(), Times.Once());
+                microsoftSQLMock.Verify(m => m.GetDurabilityProvider(), Times.Never());
+            }
+        }
 
         [Theory]
         [Trait("Category", PlatformSpecificHelpers.TestCategory)]
-        [MemberData(nameof(ValidStorageProviderTypesData))]
-        public void SelectingAvailableStorageProvider(string storageProvider, IDurabilityProviderFactory expectedFactory)
+        [InlineData("MicrosoftSQL")]
+        public void SelectingMicrosoftSQLStorageProvider(string storageProvider)
         {
-            IDurabilityProviderFactory defaultFactory = AllDurabilityProviderFactories.First(f => f.Name.Equals(storageProvider));
+            var orchestrationServiceClientMock = new Mock<IOrchestrationServiceClient>();
+            Mock<IDurabilityProviderFactory> azureStorageMock = GetAzureStorageStorageProviderMock(orchestrationServiceClientMock);
+            Mock<IDurabilityProviderFactory> microsoftSQLMock = GetMicrosoftSQLStorageProviderMock(orchestrationServiceClientMock);
+            Mock<IDurabilityProviderFactory> netheriteMock = GetNetheriteStorageProviderMock(orchestrationServiceClientMock);
+
+            IEnumerable<IDurabilityProviderFactory> durabilityProviderFactories = new[] { azureStorageMock.Object, microsoftSQLMock.Object, netheriteMock.Object };
+
+            DurableTaskOptions options = new DurableTaskOptions();
+            options.StorageProvider["type"] = storageProvider;
 
-            Assert.Equal(expectedFactory, defaultFactory);
+            using (ITestHost host = TestHelpers.GetJobHostWithMultipleDurabilityProviders(
+                options: options,
+                durabilityProviderFactories: durabilityProviderFactories))
+            {
+                azureStorageMock.Verify(a => a.GetDurabilityProvider(), Times.Never());
+                netheriteMock.Verify(n => n.GetDurabilityProvider(), Times.Never());
+                microsoftSQLMock.Verify(m => m.GetDurabilityProvider(), Times.Once());
+            }
         }
 
-        [Theory]
+        [Fact]
         [Trait("Category", PlatformSpecificHelpers.TestCategory)]
-        [InlineData("AzureStorage")]
-        public void SelectingFromEmptyFactoryListThrowsException(string storageProvider)
+        public void SelectingFromEmptyFactoryListThrowsException()
```

Comment:
Maybe we could generalize this test to `NoProviderSpecified_AzureStorageFactoryNotRegisterd()`, and then parameterize it with both the empty list and a list with netherite and sql?